### ISSUE TITLE
umb-control-group label-for property does not populate for attribute

### ISSUE THREAD

**Issue by lordscarlet** [umbraco/Umbraco-CMS#8704]
When using the `umb-control-group` directive, there is a `label-for` attribute. However, this attribute does not populate the `for` attribute of the `<label>` that is created as part of the directive. 

## Umbraco version

I am seeing this issue on Umbraco version: 8.6.1

Reproduction
------------

### Specifics

Navigate to `/umbraco/#/users/users/users?create=true`

The label/input fields do not have a `for` mapping between one another

![image](https://user-images.githubusercontent.com/17951/90895762-a0281a80-e390-11ea-8e9c-7540e82a8753.png)

Looking at the source, I see a couple of problems:

1. In the `umbControlGroup` directive, there is a `labelFor` attribute. However, that is not used for the `<label>` element's `for` attribute. Instead it uses the `alias` property to set a value on the `for` attribute. https://github.com/umbraco/Umbraco-CMS/blob/55c37e0a4efacfce48c3d776b2893dd076625049/src/Umbraco.Web.UI.Client/src/views/components/html/umb-control-group.html#L4
2. Even if `alias` was used when using the `umb-control-group` directive, or (1) was not an issue, in many (all?) instances of the `umb-control-group` directive, the related `<input>` does not have an id, so the `for` attribute would not have any value regardless https://github.com/umbraco/Umbraco-CMS/blob/55c37e0a4efacfce48c3d776b2893dd076625049/src/Umbraco.Web.UI.Client/src/views/users/views/users/users.html#L318-L319
### Expected result

1. The `<label>` should have a `for` attribute with the same value as the `id` of the `<input>` field
2. The `<input>` field should have an `id` attribute that matches the `for` in the `<label>`

### Actual result

1. The `<label>` does not have a value for the `for` attribute
2. The `<input>` does not have an `id` attribute to relate to the `for` attribute

**Comment by MMasey** [comment]
@nul800sebastiaan Can this be tagged with the accessibility label please. Thanks

**Comment by nul800sebastiaan** [comment]
Thanks @lordscarlet ! I see you've added a few issues around accessibility and I'll label them as such and put them in `up for grabs`. I don't know if you intended to work on them or just report them? If you intend to help out with these issue then we're more than happy to look at a PR without having to create an issue for each of them, just so you know! 👍

**Comment by nul800sebastiaan** [comment]
Additionally, if you're not aware yet, we have an accessibility team who are coordinating more accessibility updates. More info and links to their Trello board etc can be found here: https://our.umbraco.com/get-involved/the-accessibility-team/

**Comment by lordscarlet** [comment]
No problem at all @nul800sebastiaan. I will probably address a lot of them, but I am getting them in to start. We currently have an edict that we can't onboard any users until the 508 compliance is improved.

I will check out your link, but also complain at the same time: It is a little tough for someone coming in and trying to see the status of accessibility issues if you have a separate, not obvious location where accessibility issues are cataloged. I went to the regular github issues expecting to find all known issues here.

**Comment by nul800sebastiaan** [comment]
I understand @lordscarlet! The idea is that there is a LOT of stuff that needs improving and the team didn't want to frustrate the process we have on the issue tracker too much, which is we "approve" all incoming issues. We all agree that these issues need to be fixed and we're happy to take any and all incoming contributions for that! I am just sorry I didn't catch you earlier after the first few issues, would have immediately pointed you in that direction! 👍

**Comment by lordscarlet** [comment]
Not to belabor the point (but obviously I am about to :) ), but someone evaluating the status of accessibility in Umbraco could take a look at the current release, see there are too many problems, check the issues on github, and decide nothing is being done about it and move on to another solution. 

Thank you for pointing me in the right direction!

**Comment by nul800sebastiaan** [comment]
I'm seeing a great deal of activity, just need a hint as to where to look I guess 😅 https://github.com/umbraco/Umbraco-CMS/issues?q=label%3Acategory%2Faccessibility+

**Comment by lordscarlet** [comment]
I went there, and I looked for open issues, which is a much smaller bucket. However, I missed this one, which would have helped some. https://github.com/umbraco/Umbraco-CMS/issues/5277

Again, not trying to get into any sort of argument here, but I hope you can see the frustration with having two different places to look for open issues.

**Comment by umbrabot** [comment]
Hiya @lordscarlet,

Just wanted to let you know that we noticed that this issue got a bit stale and might not be relevant any more.

We will close this issue for now but we're happy to open it up again if you think it's still relevant (for example: it's a feature request that's not yet implemented, or it's a bug that's not yet been fixed). 

To open it this issue up again, you can write `@umbrabot still relevant` in  a new comment as the first line. It would be *super* helpful for us if on the next line you could let us know why you think it's still relevant.

For example:

>@umbrabot still relevant
>This bug can still be reproduced in version `x.y.z` 

This will reopen the issue in the next few hours.

Thanks, from your friendly Umbraco GitHub bot :robot: :slightly_smiling_face:
